### PR TITLE
Update participant dashboard visualizations with group specifics.

### PR DESCRIPTION
**Pull request by eschlange**
* Added group expectation to controller methods.
* Updated routes.
* Updated patient vis helper to include group specification.
* Updated views that use the path helpers.
* Added specs to check for error when accessed by withdrawn participants.

[#118113669]

### CODE REVIEW

File: app/controllers/think_feel_do_engine/coach/participant_activities_visualizations_controller.rb [hunk 1]
```diff
@@ -12,6 +12,7 @@ class ParticipantActivitiesVisualizationsController < ApplicationController
 
       def show
         @participant = Participant.find(params[:participant_id])
+        @group = Group.find(params[:group_id])
```
Comment (michaelwehrley) [inline app/controllers/think_feel_do_engine/coach/participant_activities_visualizations_controller.rb]: I think we should add a `rescue` here if `Group` can't be found.  Does that make sense?

File: app/controllers/think_feel_do_engine/coach/participant_thoughts_visualizations_controller.rb [hunk 2]
```diff
@@ -8,6 +8,7 @@ class ParticipantThoughtsVisualizationsController < ApplicationController
 
       def show
         @participant = Participant.find(params[:participant_id])
+        @group = Group.find(params[:group_id])
```
Comment (michaelwehrley) [inline app/controllers/think_feel_do_engine/coach/participant_thoughts_visualizations_controller.rb]: And here.

File: app/helpers/think_feel_do_engine/coach/patient_dashboard_helper.rb [hunk 3]
```diff
@@ -11,7 +11,7 @@ def breadcrumbs
         return unless VISUALIZATION_CONTROLLERS.include?(controller_name)
 
         dashboard_path = coach_group_patient_dashboard_path(
-          @participant.active_group,
+          @group,
```
Comment (michaelwehrley) [inline app/helpers/think_feel_do_engine/coach/patient_dashboard_helper.rb]: Would it make sense to maybe actually find the group off `@participant.groups.find(params[:group_id])`

File: app/controllers/think_feel_do_engine/coach/participant_activities_visualizations_controller.rb (hunk 1 above)
Comment (michaelwehrley) [inline app/controllers/think_feel_do_engine/coach/participant_activities_visualizations_controller.rb]: After looking this over, I think we should find group based off `@participant.groups.find(params[:group_id])` and include a `rescue` in both this and `participant_thoughts_visualizations_controller`.

Comment (eschlange) [inline app/controllers/think_feel_do_engine/coach/participant_activities_visualizations_controller.rb]: I'll add a catch-all for participant and group.

File: app/helpers/think_feel_do_engine/coach/patient_dashboard_helper.rb (hunk 3 above)
Comment (eschlange) [inline app/helpers/think_feel_do_engine/coach/patient_dashboard_helper.rb]: No, a withdrawn participant lists the group in that scope.

**Comment by michaelwehrley** [comment]
♐ Good work!

**Pull request merged**